cli/src/parse.rs

```diff
@@ -393,6 +431,99 @@ pub fn parse_file_at_path(parser: &mut Parser, opts: &ParseFileOptions) -> Resul
     })
 }
 
+fn cst_render_node(
+    opts: &ParseFileOptions,
+    cursor: &mut TreeCursor,
+    source_code: &[u8],
+    stdout: &mut StdoutLock<'static>,
+    row_width: usize,
+    indent_level: usize,
+) -> Result<()> {
+    let node = cursor.node();
+    let is_named = node.is_named();
+    let start = node.start_position();
+    let end = node.end_position();
+    if !opts.no_ranges {
+        let range_color = if is_named && cursor.field_name().is_none() {
+            Some(AnsiColor::BrightYellow)
+        } else {
+            Some(AnsiColor::White)
+        };
+        write!(
+            stdout,
+            "{}",
+            paint(
+                range_color,
+                &format!(
+                    " {:row_width$}:{:<3} - {:row_width$}:{:<3}",
+                    start.row, start.column, end.row, end.column,
+                )
+            )
+        )?;
+    }
+    write!(stdout, "{}", "  ".repeat(indent_level))?;
+    if is_named {
+        if let Some(field_name) = cursor.field_name() {
+            write!(
+                stdout,
+                "{}",
+                paint(Some(AnsiColor::Blue), &format!("{field_name}: "))
+            )?;
+        }
+        let kind_color = if node.kind().eq("ERROR") {
+            Some(AnsiColor::Red)
+        } else {
+            Some(AnsiColor::BrightCyan)
+        };
+        write!(stdout, "{} ", paint(kind_color, node.kind()))?;
+        if node.child_count() == 0 {
+            if let Ok(text) = node.utf8_text(source_code) {
+                write!(stdout, "`{}`", paint(Some(AnsiColor::BrightGreen), text))?;
+            } else {
+                write!(
+                    stdout,
+                    "{}",
+                    paint(
+                        Some(AnsiColor::Red),
+                        &String::from_utf8_lossy(&source_code[node.start_byte()..node.end_byte()])
+                    )
+                )?;
+            }
+        }
+        writeln!(stdout)?;
+    } else if node.is_missing() {
+        writeln!(
+            stdout,
+            "{}",
+            &format!(
+                "{}: \"{}\"",
+                paint(Some(AnsiColor::Red), "MISSING"),
+                node.kind()
+            )
+        )?;
+    } else if let Ok(text) = node.utf8_text(source_code) {
+        writeln!(
+            stdout,
+            "\"{}\"",
+            paint(
+                Some(AnsiColor::Green),
+                &text.replace('\n', "\\n").to_string()
```

Comment:
If the text spans multiple lines, the renderer looks pretty ugly with it intruding on row info. I think each line should be processed separately.